### PR TITLE
feat(gatsby) : add createContentDigest helper

### PR DESCRIPTION
**Pull request by rbadr**
This PR add a createContentDigest helper and pass it in to the Gatsby API runner.

Closes https://github.com/gatsbyjs/gatsby/issues/8587

### CODE REVIEW

File: packages/gatsby/src/utils/__tests__/create-content-digest.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+describe(`Create content digest`, () => {
+  it(`returns the content digest`, () => {
+    const content = JSON.stringify({ id: 1 })
+
+    const contentDigest = createContentDigest(content)
+
+    expect(contentDigest).toEqual(`d2ce28b9a7fd7e4407e2b0fd499b7fe4`)
+  })
+
+  it(`throws an error when the data argument is not a string`, () => {
```
Comment (pieh) [inline packages/gatsby/src/utils/__tests__/create-content-digest.js]: Could we actually handle non strings args? One less thing that plugin authors need to worry about

Comment (KyleAMathews) [reply packages/gatsby/src/utils/__tests__/create-content-digest.js]: Yeah just stringify whatever comes in.

**Comment by KyleAMathews** [comment]
Oops didn't mean to resolve that discussion. Can't seem to unresolve it from my phone.

Also, we should move some at least of the plugins over to use this to further test that this change is working correctly plus show real world examples of using it.

File: packages/gatsby/src/utils/__tests__/create-content-digest.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const createContentDigest = require(`../create-content-digest`)
+
+describe(`Create content digest`, () => {
+  it(`returns the content digest`, () => {
+    const content = JSON.stringify({ id: 1 })
+
+    const contentDigest = createContentDigest(content)
+
+    expect(contentDigest).toEqual(`d2ce28b9a7fd7e4407e2b0fd499b7fe4`)
```
[the remark this answers is not in the document]
Comment (rbadr) [reply packages/gatsby/src/utils/__tests__/create-content-digest.js]: Yes, I agree !

File: packages/gatsby/src/utils/__tests__/create-content-digest.js (hunk 1 above)
Comment (rbadr) [reply packages/gatsby/src/utils/__tests__/create-content-digest.js]: That's a good point

**Comment by KyleAMathews** [comment]
It doesn't matter which ones. Just pick 3-4 at random and make the change. We'll actually pick ones with good tests to make sure the change doesn't have any effect on the plugin tests.

File: packages/gatsby-transformer-xml/src/gatsby-node.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ const parseXml = require(`xml-parser`)
 const crypto = require(`crypto`)
```
Comment (pieh) [inline packages/gatsby-transformer-xml/src/gatsby-node.js]: please remove this import ( https://circleci.com/gh/gatsbyjs/gatsby/6659?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link )

Comment (rbadr) [reply packages/gatsby-transformer-xml/src/gatsby-node.js]: my bad

**Comment by rbadr** [comment]
The travis ci is failing, did I miss something ?

**Comment by pieh** [comment]
> The travis ci is failing, did I miss something ?

Travis is flaky sometimes - don't worry about that

**Comment by DSchau** [comment]
```
Successfully published:
 - gatsby-transformer-csv@2.0.1
 - gatsby-transformer-json@2.1.3
 - gatsby-transformer-remark@2.1.6
 - gatsby-transformer-xml@2.0.1
 - gatsby@2.0.15
```

`gatsby@2.0.15` is the important one here :)

**Pull request merged**